architecture.rst

```diff
@@ -121,14 +121,29 @@ Netmaker can be used in its entirety without the UI, but the UI makes things a l
 CoreDNS
 --------
 
-As of 0.12.0, CoreDNS is not an active part of the Netmaker system. Nodes DO NOT receive their DNS updates using a nameserver. Instead, DNS entries are added to the local "hosts" file directly. The CoreDNS component can be safely removed from the setup and DNS will continue to function.
+As of 0.22.0, CoreDNS is an active part of the Netmaker system. We deprecated setting entries on the hosts file which was not a ideal implementation.
+Netmaker server actively sets the dns entries on the CoreDNS server.
+After you install the netmaker server components, you can see the corendns container running as well.
+User need to make some changes manually to activate the corendns server, follow these steps on the netmaker server :-
+1. disable the systemd-resolved (Reason: to avoid port conflict with coredns server )
+.. code-block::
 
-Previously, CoreDNS was used as a nameserver and the netclient would set the nameserver per peer. However, this only worked on a subset of Linux systems, because it required resolvectl. The new method (using the hosts file) works across OSs.
+   sudo systemctl disable systemd-resolved.service
+   sudo systemctl stop systemd-resolved
 
-However, we still maintain CoreDNS in the default deployment for two reasons:  
-  1. You may wish to add a nameserver to "Ext Clients". CoreDNS can still be used for this.  
-  2. You may wish to integrate the Netmaker nameserver with your existing DNS setup.  
+2. Uncomment the `network_mode: host` on the coredns container spec in `/root/docker-compose.yml` and run `docker-compose up -d`
 
+3. `**IMPORTANT:**` Since you have disabled systemd-resolved service on server, make sure to set the nameserver to public ip of the machine, which basically points to the coredns server.
```

Comment:
are these steps for all users or for only those upgrading to v0.22?

Reply:
This is in general for all users who wants to setup coredns